src/app/data-model-provider/MetadataTypes.h

```diff
@@ -140,18 +167,26 @@ class ProviderMetadataTree
 public:
     virtual ~ProviderMetadataTree() = default;
 
-    virtual EndpointId FirstEndpoint()                 = 0;
-    virtual EndpointId NextEndpoint(EndpointId before) = 0;
+    virtual EndpointEntry FirstEndpoint()                              = 0;
+    virtual EndpointEntry NextEndpoint(EndpointId before)              = 0;
```

Comment:
If we start having information, the other first/next generally contain the "info" in the return structure. I would consider making the methods look the same in this case.